### ISSUE TITLE
yyjson_mut_write supported pass char* dest to avoid memory copy

### ISSUE THREAD

**Issue by gubaojian** [ibireme/yyjson#184]
currently, cann't pass dest buffer : 
yyjson_mut_write(doc, 0, &len);

can yyjson_mut_write supported dest buffer
char* dest; // buffer
yyjson_mut_write(doc, 0, &len, dest);

**Comment by ibireme** [comment]
You're right, this is a missing feature. It shouldn't be too hard to add.

One thing to note: the required buffer size may be larger than the final JSON length. For example, serializing "abc" needs 5 bytes (3 + 2), but the JSON writer would need to reserve 20 bytes (3 * 6 + 2) of space.

I'll get this implemented soon.
